File: 0034-find-first-and-last-position-of-element-in-sorted-array/solution.java
class Solution {
    public int[] searchRange(int[] nums, int target) {
        int[] ans=new int[2];
        ans[0]=binsearch(nums,target,true);
        ans[1]=binsearch(nums,target,false);
        return ans;
    }
    int binsearch(int [] nums,int target,boolean findsearchindex){
        int start=0;
        int end=nums.length-1;
        int ans=-1;
        
        while(start<=end){
            int mid=start+(end-start)/2;
            if(target<nums[mid]){
                end=mid-1;
            }
            else if(target>nums[mid]){
                start=mid+1;
            }
            else{
                ans=mid;
                if(findsearchindex==true){
                    end=mid-1;
                }
                else{
                    start=mid+1;
                }
            }
        }
        return ans;
    }
}

File: 0034-find-first-and-last-position-of-element-in-sorted-array/solution.py
class Solution:
    def searchRange(self, nums: List[int], target: int) -> List[int]:
        def first(nums,target):
            start=0
            end=len(nums)-1
            first=-1
            while(start<=end):
                mid=start+(end-start)//2
                if(nums[mid]==target):
                    first=mid
                    end=mid-1
                elif(nums[mid]<target):
                    start=mid+1
                else:
                    end=mid-1
            return first
        def last(nums,target):
            start=0
            end=len(nums)-1
            last=-1
            while(start<=end):
                mid=start+(end-start)//2
                if(nums[mid]==target):
                    last=mid
                    start=mid+1
                elif(nums[mid]<target):
                    start=mid+1
                else:
                    end=mid-1
            return last
        return [first(nums,target),last(nums,target)]

